typeclasses/maps.py

```diff
@@ -0,0 +1,237 @@
+# -*- encoding: utf-8 -*-
+
+from items import Item
+from utils.directed_exits import get_directed_exits, offset_for_exit
+from utils.search_parents import find_room, find_character
+
+TILES = {
+    'WATER': u'≈≈≈',
+    'DEEPWATER': u'≋≋≋',
+    'VEGETATION': u'~↟~',
+    'GRASS': u' . ',
+}
+CELL_WIDTH = 3
+PADDING = 1
+
+class Map(Item):
+    '''
+    The Map typeclass creates an empty map which fills up
+    as its owner travels through Ainneve.
+    When looking at the map, a representation of the world
+    as explored so far is shown.
+
+    Rooms can customize the map tile drawn for them in several ways:
+        - by specifying a `map_tile()` function that takes the requesting map
+          and returns a three-character string to draw on the map
+        - by specifying a `map_tile` property containing
+          a three-character string to draw on the map
+        - by specifying a `map_tile` Attribute containing
+          a three-character string to draw on the map
+        - by specifying a `terrain` present in the TILES dictionary in maps.py
+    If no tile can be found, the map will display `?!?` as the tile.
+    '''
+    weight = 1.0
+
+    def at_object_creation(self):
+        super(Map, self).at_object_creation()
+        self.db.map_data = {}
+        self.db.x = 0
+        self.db.y = 0
+        self.db.off_map = False
+        self.map_current_room()
+
+    @property
+    def room(self):
+        '''
+        Finds the room that this object is currently in,
+        or None if it isn’t in a room.
+        This still works if the map is inside one or more nested objects.
+        '''
+        return find_room(self)
+
+    @property
+    def current_character(self):
+        '''
+        This returns the character carrying the map,
+        or None if the map isn’t currently being carried by a character.
+        '''
+        return find_character(self)
+
+    def draw_map(self, show_player = True):
+        '''
+        This returns a two-dimensional list containing the map tiles.
+
+        If the owner is a builder, the map displays rows
+        containing the DB numbers for each room above each row.
+
+        If the show_player argument is `True`, the room the owning player
+        is currently in will have its middle character replaced with a `*`.
+        '''
+        player = self.current_character.player
```

Comment:
Ah, missed this. It should be current_character.account - since the name changed.